### ISSUE TITLE
No dividers between preferences?

### ISSUE THREAD

**Issue by riyaz** [Gericop/Android-Support-Preference-V7-Fix#5]
With suppor lib preferencefragment, we don't see any dividers in between preferences, I thought it should have been fixed in your workaround, its not.

is it normal?

**Comment by Gericop** [comment]
The dividers are gone because of the more generic nature of the RecyclerView. The workaound consist of using ItemDecoration to draw (on to the actual canvas) the dividers. I did some feasability tests and yes, it's very feasible, however, the method, in my opinion, uses too much CPU / GPU as it redraws these dividers for every single child every time the list changes position. Also, I did not consider the lack of dividers a serious bug, compared to the others though.

**Comment by BraisGabin** [comment]
I *fix* this. You just need to add this code to your `PreferenceFragmentCompat`:
```java
  @Override
  public void onViewCreated(View view, Bundle savedInstanceState) {
    ((RecyclerView) view.findViewById(R.id.list)).addItemDecoration(
        new HorizontalDividerItemDecoration.Builder(getActivity())
            .showLastDivider()
            .build()
    );
  }
```
I'm using the library [RecyclerView-FlexibleDivider](https://github.com/yqritc/RecyclerView-FlexibleDivider).

**Comment by stschwark** [comment]
@BraisGabin, at first I also just tried to add a generic ItemDecoration for the dividers. What I found though was that the preference category headers are rows in the list too. As a consequence you end up with dividers above and below those category headers which looks bad. Have you found an easy way around that without baking knowledge about the internals of the preference list into the ItemDecoration?
